src/steps/resolve-imports.ts

```diff
@@ -0,0 +1,652 @@
+            'cannot use {{component}} keyword for string-based resolution. ' +
+            'If it is only used for arguments currying, that is fine, ' +
+            'but this codemod is not currently sophisticated enough to ' +
+            'tell the difference.',
+        );
+      }
+
+      const resolved = resolver.resolve('Append', node.path);
+
+      if (resolved) {
+        if (node.path.original !== resolved.identifier) {
+          node.path = AST.builders.path(resolved.identifier);
+        }
+      }
+    },
+
+    SubExpression(node) {
+      // (true), etc
+      if (node.path.type !== 'PathExpression') {
+        return;
+      }
+
+      if (hasActionImport && node.path.original === 'action') {
+        throw new Error(
+          'The template contains usage of the {{action}} keyword, ' +
+            'but the JS side imports { action } from @ember/object. ' +
+            'It would have shadowed the template keyword.',
+        );
+      }
+
+      if (node.path.original === 'component') {
+        throw new Error(
+          'The template contains usage of the (component) keyword, ' +
+            'this is potentially unsafe because strict mode templates ' +
+            'cannot use (component) keyword for string-based resolution. ' +
+            'If it is only used for arguments currying, that is fine, ' +
+            'but this codemod is not currently sophisticated enough to ' +
+            'tell the difference.',
+        );
+      }
+
+      const resolved = resolver.resolve('Call', node.path);
+
+      if (resolved) {
+        if (node.path.original !== resolved.identifier) {
+          node.path = AST.builders.path(resolved.identifier);
+        }
+      }
+    },
+  });
+
+  const hbsRewritten = AST.print(ast);
+  const jsRewritten = resolver.imports() + jsSource;
+
+  writeFileSync(hbs, hbsRewritten, { encoding: 'utf8' });
+  writeFileSync(gjs, jsRewritten, { encoding: 'utf8' });
+}
+
+function dasherize(camelized: string): string {
+  return camelized
+    .replace(/::/g, '/')
+    .replace(/([A-Z])/g, '-$1')
+    .replace(/^-/, '')
+    .replace(/\/-/g, '/')
+    .toLowerCase();
+}
+
+function identifierFor(dasherized: string, upperCase = false): string {
+  const parts = dasherized.split('/');
+  const name = parts[parts.length - 1];
+
+  if (!name) {
+    throw new Error('unexpected fail: ' + dasherized);
+  }
+
+  const words = name?.split('-').map((part) => {
+    return part.slice(0, 1).toUpperCase() + part.slice(1);
+  });
+
+  if (!upperCase) {
+    if (!words[0]) {
+      throw new Error('unexpected fail: ' + dasherized);
+    }
+
+    words[0] = words[0]?.toLowerCase();
+  }
+
+  return words?.join('');
+}
+
+interface ImportSpec {
+  importPath: string;
+  importType: 'default' | 'named';
+}
+
+interface Resolved extends ImportSpec {
+  identifier: string;
+}
+
+interface Search {
+  base: string;
+  module: string;
+}
+
+class Resolver {
```

Comment:
Okay, this is super nice!

I do wonder what it would take to also support third party addon imports. I suppose crawling `node_modules` would be one way.